### PR TITLE
Add `tap-stackoverflow-sampledata`

### PR DESCRIPTION
**Pull request by BuzzCutNorman**
I would like to submit tap-stackoverflow-sampledata for review as a possible addition to the Meltano Hub.  tap-stackoverflow-sampledata is a Singer tap for the Stack Overflow xml dump files avaiable at [Archieve.org](https://archive.org/details/stackexchange). This tap is inteded to be use to test Singer targets and or seed a source system with enough data to sufficiently test a source to target pipleline.

All Singer definitions are stored in `/_data/taps/` or `/_data/targets`. The minimal requirement for adding a tap or target will match the following format:

```yaml
description: General description of what the company behind the API does
entity_type: Review the JSON Schema for latest options. Currently api, file, database.
entity_url: URL of the developer documentation or website
label: Properly formatted label of the connector
name: The unique name of the connector
singer_name: The Singer specific name of the connector. Typically tap-<name> or target-<name>
type: tap or target
variants: # an array of variant types
- default: true or false
  maintenance_status: "Options: active, beta, development, inactive, unknown"
  meltano_sdk: true or false
  name: Name of the GitHub/GitLab namespace
  pip_url: git+<git_url>.git or pip instal name
  repo: repo URL
  capabilities:
    - catalog
    - discover
  settings:
    # Can be an empty array if unknown: []
    # Describe the list of supported settings, for example:
    - name: project_ids
      kind: array
      label: Project IDs
      description: Array of project IDs.
    - name: username
      label: Username
      description: Credentials used when for connecting to the source.
    - name: password
      kind: password
      label: Password
      description: Credentials used when for connecting to the source.
```

## Checklist

- [x] Add/update the file in the appropriate folder (`/taps` or `/targets`). The name of the file should match the name of the tap. If there is already one, add a descriptor to the name such as `-search`.
- [x] Add/update the PNG logo image in `/assets/logos/<taps or targets>`. The image name must match the YAML file name.
- [x] Tag `@tayloramurphy` or `@pnadolny13` to flag it for review. Or post to the [#hub](https://meltano.slack.com/archives/C01UGBSJNG5) channel on Meltano slack.

## Reviewer Checklist

- [ ] Validate file against JSON Schema. https://www.jsonschema.net/home is an option.
- [ ] Build website locally and validate everything works.

### CODE REVIEW

**Comment by BuzzCutNorman** [comment]
@pnadolny13 when you get a chance, please review this PR.  Thanks

File: _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+name: tap-stackoverflow-sampledata
+description: Tap for the Stack Overflow xml dump files avaiable at Archieve.org .  Intended as a loader plugin testing aid.
```
Comment (edgarrmondragon) [inline _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml:2]:
```suggestion
description: Tap for the Stack Overflow XML dump files available at Archive.org.  Intended as a loader plugin testing aid.
```

File: _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+namespace: tap_stackoverflow_sampledata
+logo_url: /assets/logos/extractors/tap-stackoverflow-sampledata.png
+variant: buzzcutnorman
+capabilities:
+- state
+- catalog
+- discover
+settings:
+- name: stackoverflow_data_directory
+- description: A path to the StackOverflow XML data files.
```
Comment (edgarrmondragon) [inline _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml:12]:
```suggestion
  description: A path to the StackOverflow XML data files.
```

File: _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- name: format
+- description: Currently the only format is 'jsonl'
+- name: compression
+- description: Currently the only compression options is 'gzip'
+- name: storage
+  kind: object
+- name: root
+- description: The directory you want batch messages to be placed in example 'file://test/batches'
+- name: prefix
+- description: What prefix you want your messages to have example test-batch-
```
Comment (edgarrmondragon) [inline _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml:27]:
```suggestion
- name: batch_config.encoding.format
  description: Batch file format. Currently the only format is 'jsonl'.
- name: batch_config.encoding.compression
  description: Batch file compression. Currently the only compression options is 'gzip'.
- name: batch_config.storage.root
  description: The directory you want batch messages to be placed in. For example, 
`file://test/batches`.
- name: batch_config.storage.prefix
  description: What prefix you want your messages to have. For example, `test-batch-`.
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]
For logos, those without wordmark are preferred. This would so, for example:

```xml
<svg aria-hidden="true" width="32" height="37" viewBox="0 0 32 37">
  <path d="M26 33v-9h4v13H0V24h4v9h22Z" fill="#BCBBBB"></path>
  <path
    d="m21.5 0-2.7 2 9.9 13.3 2.7-2L21.5 0ZM26 18.4 13.3 7.8l2.1-2.5 12.7 10.6-2.1 2.5ZM9.1 15.2l15 7 1.4-3-15-7-1.4 3Zm14 10.79.68-2.95-16.1-3.35L7 23l16.1 2.99ZM23 30H7v-3h16v3Z"
    fill="#F48024"></path>
</svg>
```

File: _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+settings:
+- name: stackoverflow_data_directory
+  description: A path to the StackOverflow XML data files.
+- name: batch_config.encoding.format
+  description: Batch file format. Currently the only format is 'jsonl'.
+- name: batch_config.encoding.compression
+  description: Batch file compression. Currently the only compression options is 'gzip'.
+- name: batch_config.storage.root
+  description: The directory you want batch messages to be placed in. For example, 
+`file://test/batches`.
```
Comment (edgarrmondragon) [inline _data/meltano/extractors/tap-stackoverflow-sampledata/tap-stackoverflow-sampledata.yml:19]:
```suggestion
  description: The directory you want batch messages to be placed in. For example, `file://test/batches`.
```

**Review by edgarrmondragon — CHANGES_REQUESTED** [review]

**Comment by BuzzCutNorman** [comment]
> For logos, those without wordmark are preferred. This would so, for example:
I added another logo.  Please let me know if that one is better.  If it isn't do I need a logo for the tap ?

**Comment by BuzzCutNorman** [comment]
@edgarrmondragon sorry looks like a made a mess of this.  I apologize

**Review by edgarrmondragon — APPROVED** [review]
> @edgarrmondragon sorry looks like a made a mess of this. I apologize

@BuzzCutNorman no worries, contributing docs are lacking somewhat 🙂

**Comment by BuzzCutNorman** [comment]
> @BuzzCutNorman no worries, contributing docs are lacking somewhat 🙂

Thanks for that comment.  Thank you🙏for correcting all that.

File: _data/meltano/extractors/tap-stackoverflow-sampledata/buzzcutnorman.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+- name: stackoverflow_data_directory
+  description: A path to the StackOverflow XML data files.
+- name: batch_config.encoding.format
+  description: Batch file format. Currently the only format is 'jsonl'.
+- name: batch_config.encoding.compression
+  description: Batch file compression. Currently the only compression options is 'gzip'.
+- name: batch_config.storage.root
+  description: The directory you want batch messages to be placed in. For example, `file://test/batches`.
+- name: batch_config.storage.prefix
+  description: What prefix you want your messages to have. For example, `test-batch-`.
```
Comment (tayloramurphy) [inline _data/meltano/extractors/tap-stackoverflow-sampledata/buzzcutnorman.yml:21]:
```suggestion
- name: stackoverflow_data_directory
  label: Data Directory
  description: A path to the StackOverflow XML data files.
- name: batch_config.encoding.format
  label: Encoding Format
  description: Batch file format. Currently the only format is 'jsonl'.
- name: batch_config.encoding.compression
  label: Encoding Compression
  description: Batch file compression. Currently the only compression options is 'gzip'.
- name: batch_config.storage.root
  label: Storage Root Directory
  description: The directory you want batch messages to be placed in. For example, `file://test/batches`.
- name: batch_config.storage.prefix
  label: Storage Prefix
  description: What prefix you want your messages to have. For example, `test-batch-`.
```

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**